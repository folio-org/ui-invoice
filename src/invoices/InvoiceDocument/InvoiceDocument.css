.invoiceDocumentButton {
  outline: none;
  text-decoration: underline;
}
