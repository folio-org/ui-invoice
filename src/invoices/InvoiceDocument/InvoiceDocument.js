import React from 'react';
import PropTypes from 'prop-types';

import css from './InvoiceDocument.css';

const InvoiceDocument = ({ name, downloadDocument }) => {
  return (
    downloadDocument ? (
      <button
        type="button"
        className={css.invoiceDocumentButton}
        onClick={downloadDocument}
      >
        {name}
      </button>
    ) : (
      <span>{name}</span>
    )
  );
};

InvoiceDocument.propTypes = {
  name: PropTypes.string.isRequired,
  downloadDocument: PropTypes.func,
};

export default InvoiceDocument;
