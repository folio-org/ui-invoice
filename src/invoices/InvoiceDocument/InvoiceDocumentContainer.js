import React, { useCallback } from 'react';
import PropTypes from 'prop-types';
import withRouter from 'react-router-dom/withRouter';

import { stripesConnect } from '@folio/stripes/core';
import { downloadBase64 } from '@folio/stripes-acq-components';

import { invoiceDocumentFromPropsResource } from '../../common/resources';

import InvoiceDocument from './InvoiceDocument';

const InvoiceDocumentContainer = ({ mutator, documentId, name }) => {
  const downloadDocument = useCallback(
    () => {
      mutator.invoiceDocumentDetails.GET()
        .then(invoiceDocumentData => {
          downloadBase64(name, invoiceDocumentData.contents.data);
        });
    },
    [name, mutator.invoiceDocumentDetails],
  );

  return (
    <InvoiceDocument
      name={name}
      downloadDocument={documentId && downloadDocument}
    />
  );
};

InvoiceDocumentContainer.manifest = Object.freeze({
  invoiceDocumentDetails: {
    ...invoiceDocumentFromPropsResource,
    accumulate: true,
    fetch: false,
  },
});

InvoiceDocumentContainer.propTypes = {
  mutator: PropTypes.object.isRequired,
  name: PropTypes.string.isRequired,
  documentId: PropTypes.string,
};

InvoiceDocumentContainer.defaultProps = {
  documentId: '',
};

export default withRouter(stripesConnect(InvoiceDocumentContainer));
