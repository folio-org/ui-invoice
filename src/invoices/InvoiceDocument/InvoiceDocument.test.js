import { render } from '@folio/jest-config-stripes/testing-library/react';

import InvoiceDocument from './InvoiceDocument';

const renderInvoiceDocument = (props) => (render(
  <InvoiceDocument {...props} />,
));

describe('InvoiceDocument', () => {
  beforeEach(() => {
    global.document.createRange = global.document.originalCreateRange;
  });

  afterEach(() => {
    global.document.createRange = global.document.mockCreateRange;
  });

  it('should display document name as a button when document can be dowloaded', () => {
    const { asFragment } = renderInvoiceDocument({
      name: 'Invoice Document',
      downloadDocument: jest.fn(),
    });

    expect(asFragment()).toMatchSnapshot();
  });

  it('should display document name as a text when document can not be downloaded', () => {
    const { asFragment } = renderInvoiceDocument({
      name: 'Invoice Document',
    });

    expect(asFragment()).toMatchSnapshot();
  });
});
