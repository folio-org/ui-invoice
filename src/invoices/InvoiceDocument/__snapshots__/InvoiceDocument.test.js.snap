// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvoiceDocument should display document name as a button when document can be dowloaded 1`] = `
<DocumentFragment>
  <button
    class="invoiceDocumentButton"
    type="button"
  >
    Invoice Document
  </button>
</DocumentFragment>
`;

exports[`InvoiceDocument should display document name as a text when document can not be downloaded 1`] = `
<DocumentFragment>
  <span>
    Invoice Document
  </span>
</DocumentFragment>
`;
